### ISSUE TITLE
How do I create ILoggerFactory logs with NLog?

### ISSUE THREAD

**Issue by TJSoftware** [aspnet/Logging#292]
### Summary
I am attempting to add NLog to the ILoggerFactory in a web application.  I am using a cloned copy of [your console sample](https://github.com/aspnet/Logging/tree/dev/samples) as a guide.  I have also been reading [this article](http://www.howtobuildsoftware.com/index.php/how-do/bhqn/dependency-injection-nlog-aspnet-5-add-nlog-provider-to-iloggerfactory).  When I added `Microsoft.Framework.Logging.NLog` or `Microsoft.Extensions.Logging.NLog` to the project.json neither package is not found.  I searched nuget.org to see if the package was there and it is not.  I decided to just use the package manager to add NLog and the config as I have done pre .Net 5.  I had to manually add the NLog.config file (not really a concern now).  My issue is that I am unable to include a reference that will allow me to use `loggerFactory.AddNLog` and `loggerFactory.AddEventLog`.

### Functional Impact
I am unable to create logs.

### Reproduce the Issue
Clone [https://github.com/TJSoftware/LoggingTesting.git](https://github.com/TJSoftware/LoggingTesting.git)

### My Question(s)
What is the correct NuGet package to use with ILoggerFactory `NLog` or `Microsoft.Framework.Logging.NLog`/`Microsoft.Extensions.Logging.NLog`?  If the latter, where do I find this package?

Does anyone see what else I can do to get this to work?

### Code
Here is my relavant project.json code.
```
"dependencies": {
    ...Other Deps Here...
    "NLog": "4.2.0",
    "NLog.Config": "4.2.0"
  },
  "frameworks": {
    "dnx451": {
      "dependencies": {
      }
    }
  },
  ```
	
Here is the relevant Startup.cs code
```
loggerFactory.AddNLog(new NLog.LogFactory());
loggerFactory.AddEventLog();
//This was a long shot, but worth a try
//loggerFactory.AddProvider((ILoggerProvider)new NLog.LogFactory());
```

   
### Tech Details
I am using .NET Version Manager v1.0.0-rc2-15542
I have used the runtimes
1.0.0-beta8 clr (both x64, x86) win
I am on a win 10 machine using vs 2015 enterprise.

**Comment by muratg** [comment]
Currently we're not releasing M.E.L.NLog package on nuget.  You should be able to clone the source and add it to your solution though.

Pre ASP.NET 5 methods for adding packages would not work, as you need to have all your references in project.json.

**Comment by makzr** [comment]
I agree with TJSoftware, the sample code doesn't work for me either.

I copied the files "NLogLoggerProvider.cs" and "NLogLoggerFactoryExtensions.cs" to my project and referenced NLog in my project.json:
```
  "frameworks": {
    "dnx451": {
      "dependencies": {
        "NLog": "4.2.1"
      }
    }
  },
```
The I added the following code to my Startup.cs:
```
 loggerFactory.AddNLog(new NLog.LogFactory());
```
I used the simple file logger from the samples project:
```
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      autoReload="true">
  <targets>
    <target name="logfile" xsi:type="File" fileName="log.txt"
            layout="${longdate}|${level:uppercase=true}|${logger}|${event-context:item=EventId}|${message}|${ndc}" />
  </targets>
  <rules>
    <logger name="*" minlevel="Info" writeTo="logfile" />
  </rules>
</nlog>
```
The application compiles, but the log file is not created. 
I'm using Windows, Visual Studio 2015 and ASP.NET 5 version 1.0.0-rc1-final.

**Comment by TJSoftware** [comment]
##### Summary
I was able to do some more tests since I first posted this, I should have updated the thread, but better late than never.  I was also able to get the [sample console solution](https://github.com/aspnet/Logging/tree/dev/samples) to build and run, however I do not see the `file.txt` created as the NLog config is setup for.  What I do see is the logs being displayed on the console printout.  

##### Question
Should I expect a log file `file.txt` to be created or is this an in program logging only?  I would expect the `file.txt`.

##### Sample Console at Runtime
![image](https://cloud.githubusercontent.com/assets/4823412/11475687/d7b5cc0a-973b-11e5-8f28-1fcebf4cfe0b.png)

##### NLog Config
```
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" 
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      autoReload="true">

  <targets>
    <target name="logfile" 
            xsi:type="File" 
            fileName="file.txt" 
            layout="${longdate}|${level:uppercase=true}|${logger}|${event-context:item=EventId}|${message}|${ndc}" />
    <target name="console" 
            xsi:type="ColoredConsole" 
            layout="[${level:uppercase=true}:${logger}] ${message}"/>
  </targets>

  <rules>
    <logger name="*" minlevel="Info" writeTo="logfile,console" />
  </rules>
</nlog>
```

**Comment by makzr** [comment]
Yes, you should expect a log file "file.txt" to be created.

When you do the configuration in code (see my example) the log file is created in the wwwroot folder.

**Comment by TJSoftware** [comment]
@makzr Yes, adding the configuration in cs code provides me with a file now.  I made the changes to the sample console project you mentioned as well as update to NLogLoggerProvider.  There was a reference to an obsolete method.  I added the IFormatProvider parameter and moved the exception to the new parameter location.  Thank you for the solution.

```
IFormatProvider formatProvider = new System.Globalization.CultureInfo("en-US");
var eventInfo = LogEventInfo.Create(nLogLogLevel, _logger.Name, exception, formatProvider, message);
```

This gives me a better understanding of why my previous code was not working.  My ConfigurationBuilder was setup like this...

```
var loggingConfiguration = new ConfigurationBuilder().AddJsonFile("logging.json").Build();
```

Because the ConfigurationBuilder is set to read its settings from `logging.json`, the `NLog.config` does not appear to be read in.  Looking at the intellisense options I see that `ConfigurationBuilder` has `.Add(IConfigurationProvider)`, `AddInMemoryCollection(ConfigurationBuilder`, and `AddJsonFile(string)`.

>Currently we're not releasing M.E.L.NLog package on nuget. You should be able to clone the source and add it to your solution though.

@muratg You mentioned that `M.E.L.NLog` is not currently being release on NuGet.  Is this something that is a temporary delay or has this functionality been deprecated?
